src/lib/schedulerJobs/refreshInMemoryCache.ts

```diff
@@ -0,0 +1,36 @@
+import { Job } from '@hokify/agenda';
+import { Logger, getInMemoryCache, getScheduler, refreshAllPrefixCommandCategoriesCache, refreshAllPrefixCommandChannelDefaultVersionsCache, refreshAllPrefixCommandVersionsCache, refreshAllPrefixCommandsCache } from '../index';
+
+export async function refreshInMemoryCache(job: Job) {
+    const scheduler = getScheduler();
+    if (!scheduler) {
+        Logger.error('Failed to get scheduler instance');
+        return;
+    }
+
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) {
+        Logger.error('Failed to get in-memory cache instance');
+        return;
+    }
+
+    // Needed because of https://github.com/agenda/agenda/issues/401
+    // eslint-disable-next-line no-underscore-dangle
+    const matchingJobs = await scheduler.jobs({ _id: job.attrs._id });
+    if (matchingJobs.length !== 1) {
+        Logger.debug('Job has been deleted already, skipping execution.');
+        return;
+    }
+
+    const start = new Date().getTime();
+    try {
+        await refreshAllPrefixCommandVersionsCache();
+        await refreshAllPrefixCommandCategoriesCache();
+        await refreshAllPrefixCommandsCache();
+        await refreshAllPrefixCommandChannelDefaultVersionsCache();
```

Comment:
There is a risk for a race condition, I think. But it doesn't matter in what order caches are updated... The scenario is this:
Situation: DB Has a new version, and content for the new version for several commands. MEMCACHE does not have the version or content.

1. Cache update is triggered
2. New version is updated first. Command is lagging behind
3. User triggers command for that version (somehow they know it exists)
4. Handler picks up Version exists - But command has no content for it because command cache hasn't been loaded yet.
>> This should result in the handler to fall back on generic or not show anything

There's similar situations that can happen at different times, I guess. But I think most things should be captured somewhere. 

My biggest worry is what happens if someone updates a command, content, ... at the same time as the cache update runs. _WORST_ case:
* T0: Cache refresh DB Query pulls data of command content with `OLDCONTENT`
* T0+1MS: User Saves `NEWCONTENT`, DB is updated, Caches is updated
* T0+10MS: Cache refresh updates cache back with `OLDCONTENT`

BUT, the thing here is, only the cache is poisoned... There's two possibilities:
* User doesn't notice, next cache refresh pulls `NEWCONTENT` from DB, and updates cache
* User does notice, does the change again, all good. Or does a manual cache refresh to pull it.

This is the important factor of the cache, it's one direction: DB ==> CACHE. _NEVER_ CACHE ==> DB...

Reply:
BTW, cache refreshes take <0.5ms... While there is some risk for a missed cache update, I feel like it'll be manageable...

Reply:
Absolutely agree! It's totally not worth the effort.